x-pack/plugins/actions/server/actions_client.ts

```diff
@@ -729,6 +753,124 @@ export class ActionsClient {
   public isPreconfigured(connectorId: string): boolean {
     return !!this.preconfiguredActions.find((preconfigured) => preconfigured.id === connectorId);
   }
+
+  public async getGlobalExecutionLogWithAuth({
+    dateStart,
+    dateEnd,
+    filter,
+    page,
+    perPage,
+    sort,
+    namespaces,
+  }: GetGlobalExecutionLogParams): Promise<IExecutionLogResult> {
+    this.logger.debug(`getGlobalExecutionLogWithAuth(): getting global execution log`);
+
+    const authorizationTuple = {} as KueryNode;
+    try {
+      await this.authorization.ensureAuthorized('get_execution_log');
+    } catch (error) {
+      this.auditLogger?.log(
+        connectorAuditEvent({
+          action: ConnectorAuditAction.GET_GLOBAL_EXECUTION_LOG,
+          error,
+        })
+      );
+      throw error;
+    }
+
+    this.auditLogger?.log(
+      connectorAuditEvent({
+        action: ConnectorAuditAction.GET_GLOBAL_EXECUTION_LOG,
+      })
+    );
+
+    const dateNow = new Date();
+    const parsedDateStart = parseDate(dateStart, 'dateStart', dateNow);
+    const parsedDateEnd = parseDate(dateEnd, 'dateEnd', dateNow);
+
+    const eventLogClient = await this.getEventLogClient();
+
+    try {
+      const aggResult = await eventLogClient.aggregateEventsWithAuthFilter(
+        'action',
+        authorizationTuple,
+        {
+          start: parsedDateStart.toISOString(),
+          end: parsedDateEnd.toISOString(),
+          aggs: getExecutionLogAggregation({
+            filter: filter ? `${filter} AND (${ACTION_FILTER})` : ACTION_FILTER,
```

Comment:
I believe this filter should move up one level so that it is turned into a query. Passing this filter into the `getExecutionLogAggregation` function will add the filter into the aggregation but we want to only aggregate over filtered results.

```
{
          start: parsedDateStart.toISOString(),
          end: parsedDateEnd.toISOString(),
          filter: filter ? `${filter} AND (${ACTION_FILTER})` : ACTION_FILTER,
          aggs: getExecutionLogAggregation({
            page,
            perPage,
            sort,
          }),
},
```

Would that work?

Reply:
This throws an error when I move it up a level. I am wondering if we should keep it in `getExecutionLogAggregation`